addons/account/wizard/account_move_send.py

```diff
@@ -0,0 +1,464 @@
+        * id:               str: The (fake) id of the attachment, this is needed in rendering in t-key.
+        * name:             str: The name of the attachment.
+        * mimetype:         str: The mimetype of the attachment.
+        * placeholder       bool: should be true to prevent download / deletion
+        """
+        self.ensure_one()
+        filename = self._get_invoice_pdf_report_filename(move)
+        return [{
+            'id': f'placeholder_{filename}',
+            'name': filename,
+            'mimetype': 'application/pdf',
+            'placeholder': True,
+        }]
+
+    def _get_default_mail_attachments_data(self, mail_template, move):
+        """ Returns all the placeholder data and mail template data
+        """
+        results = []
+
+        for attachment in mail_template.attachment_ids:
+            results.append({
+                'id': attachment.id,
+                'name': attachment.name,
+                'mimetype': attachment.mimetype,
+                'placeholder': False,
+            })
+
+        return self._get_placeholder_mail_attachments_data(move) + results
+
+    # -------------------------------------------------------------------------
+    # BUSINESS ACTIONS
+    # -------------------------------------------------------------------------
+
+    @api.model
+    def _get_invoice_pdf_report_filename(self, move):
+        return f"{move.name.replace('/', '_')}.pdf"
+
+    @api.model
+    def _need_document(self, move):
+        return not move.invoice_pdf_report_id and move.state == 'posted'
+
+    @api.model
+    def _download(self, move):
+        """ Download the PDF. """
+        return {
+            'type': 'ir.actions.act_url',
+            'url': f'/web/content/{move.invoice_pdf_report_id.id}?download=true',
+            'close_on_report_download': True,
+        }
+
+    @api.model
+    def _send_mail(self, mail_template, move, **kwargs):
+        """ Send the journal entries passed as parameter by mail. """
+        partner_ids = kwargs.get('partner_ids', [])
+
+        move\
+            .with_context(
+                no_new_invoice=True,
+                mail_notify_author=self.env.user.partner_id.id in partner_ids,
+                mailing_document_based=True)\
+            .message_post(
+                message_type='comment',
+                **kwargs,
+                **{
+                    'email_layout_xmlid': 'mail.mail_notification_layout_with_responsible_signature',
+                    'email_add_signature': not mail_template,
+                    'mail_auto_delete': mail_template.auto_delete,
+                    'mail_server_id': mail_template.mail_server_id.id,
+                    'reply_to_force_new': False,
+                },
+            )
+        move.is_move_sent = True
+
+    def _send_mails(self, moves):
+        subtype = self.env.ref('mail.mt_comment')
+        mail_template = self.mail_template_id
+
+        for move in moves:
```

Comment:
Isn't it batched by partner?

Reply:
@william-andre No

Reply:
Doesn't mean that you cannot send one single mail with multiple attachments

Reply:
@william-andre As far I know, that wasn't possible in the previous version. Do you have something in mind allowing to do that? I hope not. Otherwise we are clearly removing a big feature here :sweat:

Reply:
From quickly reading the code of the mail composer, I don't see it being done there.
Maybe something to implement later on our side?